services/models.py

```diff
@@ -159,7 +177,7 @@ def delete_gdpr_data(self, api_token: str, dry_run=False):
             data["dry_run"] = True
 
         if self.service.gdpr_url:
-            url = urllib.parse.urljoin(self.service.gdpr_url, str(self.profile.pk))
+            url = self.service.get_gdpr_url_for_profile(self.profile)
```

Comment:
The URL generation could be before the `if` clause. Let the `Service` generate the URL and make decisions only based on the result.

Reply:
True.